src/basic/xattr-util.c

```diff
@@ -156,6 +155,10 @@ int getxattr_at_malloc(
 
                 r = getxattr_pinned_internal(fd, path, at_flags, by_procfs, name, v, l);
                 if (r >= 0) {
+                        /* Refuse extended attributes with embedded NUL bytes, they are definitely not regular strings */
```

Comment:
why though? xattr can be arbitrary data, and we append one extra NUL here only for safety. The exact number of bytes read is returned in r...

Reply:
true.

Reply:
we should maybe return the size as return param, so that we know if caller will use it or not.